lightning/src/ln/channelmanager.rs

```diff
@@ -1229,6 +1229,55 @@ impl ChannelDetails {
 	pub fn get_outbound_payment_scid(&self) -> Option<u64> {
 		self.short_channel_id.or(self.outbound_scid_alias)
 	}
+
+	fn from_channel<Signer: ChannelSigner + WriteableEcdsaChannelSigner>(channel: &Channel<Signer>,
```

Comment:
FYI, WriteableEcdsaChannelSigner inherits ChannelSigner. That said, absolutely love this method